InfoHandler.cpp

```diff
@@ -83,61 +83,63 @@ void InfoHandler::applyMethodType(BinaryViewRef bv, const ObjectiveNinja::ClassI
+
+        } else {
+            type = Type::NamedType(bv, nameOrType.name);
+
+            if (!type) {
+                BNLogWarn("Unable to resolve type info for type %s", nameOrType.name.GetString().c_str());
+                type = Type::PointerType(bv->GetAddressSize(), Type::VoidType());
+            }
+        }
+
+        return type;
     };
 
-    // Build the type string for the method.
-    std::string typeString;
-    for (size_t i = 0; i < typeTokens.size(); ++i) {
+    BinaryNinja::QualifiedNameAndType nameAndType;
+    std::set<BinaryNinja::QualifiedName> typesAllowRedefinition;
+
+    auto retType = typeForQualifiedNameOrType(typeTokens[0]);
+
+    std::vector<BinaryNinja::FunctionParameter> params;
+    auto cc = bv->GetDefaultPlatform()->GetDefaultCallingConvention();
+
+    params.push_back({ "self",
+        classTypeName.IsEmpty()
+            ? BinaryNinja::Type::NamedType(bv, { "id" })
+            : BinaryNinja::Type::NamedType(bv, classTypeName),
+        true,
+        BinaryNinja::Variable() });
+
+    params.push_back({ "sel",
+        namedType(bv, "SEL"),
+        true,
+        BinaryNinja::Variable() });
+
+    for (size_t i = 3; i < typeTokens.size(); i++) {
         std::string suffix;
-        auto part = partForIndex(i);
-
-        // The underscore being used as the function name here is critically
-        // important as Clang will not parse the type string correctly---unlike
-        // the old type parser---if there is no function name. The underscore
-        // itself isn't special, and will not end up being used as the function
-        // name in either case.
-        if (i == 0)
-            suffix = " _(";
-        else if (i == typeTokens.size() - 1)
-            suffix = ")";
-        else
-            suffix = ", ";
 
-        typeString += part + suffix;
+        params.push_back({ selectorTokens.size() > i - 3 ? selectorTokens[i - 3] : "arg",
+            typeForQualifiedNameOrType(typeTokens[i]),
+            true,
+            BinaryNinja::Variable() });
     }
-    typeString += ";";
 
-    std::string errors;
-    TypeParserResult tpResult;
-    auto ok = bv->ParseTypesFromSource(typeString, {}, {}, tpResult, errors);
-    if (ok && !tpResult.functions.empty()) {
-        auto functionType = tpResult.functions[0].type;
+    auto funcType = BinaryNinja::Type::FunctionType(retType, cc, params);
 
-        // Search for the method's implementation function; apply the type if found.
-        if (auto f = bv->GetAnalysisFunction(bv->GetDefaultPlatform(), mi.implAddress))
-            f->SetUserType(functionType);
-    }
+    // Search for the method's implementation function; apply the type if found.
+    auto f = bv->GetAnalysisFunction(bv->GetDefaultPlatform(), mi.implAddress);
+    if (f) {
+        f->SetUserType(funcType);
+    } else
+        BNLogError("Processing Type for function at %llx failed", mi.implAddress);
```

Comment:
Style: Either remove the braces from the first statement or add them to the second. Preferably the former, since that matches conventions.